From 10fbf11314d00713e18096a61fa792cbf2f358af Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Sun, 9 Feb 2025 16:22:24 +0100
Subject: [PATCH 137/142] fixup! media: i2c: Add imx371 image sensor driver

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 drivers/media/i2c/imx371.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx371.c b/drivers/media/i2c/imx371.c
index 47064b9689..f62fd9d368 100644
--- a/drivers/media/i2c/imx371.c
+++ b/drivers/media/i2c/imx371.c
@@ -966,7 +966,7 @@ static int imx371_power_on(struct device *dev)
 
 	usleep_range(400, 600);
 
-	gpiod_set_value_cansleep(imx371->reset_gpio, 1);
+	gpiod_set_value_cansleep(imx371->reset_gpio, 0);
 
 	ret = clk_prepare_enable(imx371->clk);
 	if (ret) {
@@ -986,7 +986,7 @@ static int imx371_power_off(struct device *dev)
 
 	clk_disable_unprepare(imx371->clk);
 
-	gpiod_set_value_cansleep(imx371->reset_gpio, 0);
+	gpiod_set_value_cansleep(imx371->reset_gpio, 1);
 
 	regulator_bulk_disable(IMX371_NUM_SUPPLIES, imx371->supplies);
 
-- 
2.49.0

