From d2b58adc310202a5d1cdfe55832f5689fe0929f1 Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Sun, 9 Feb 2025 19:15:47 +0100
Subject: [PATCH 141/143] fixup media: i2c: Add imx376 image sensor driver

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 drivers/media/i2c/imx376.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx376.c b/drivers/media/i2c/imx376.c
index 9352e74656..e062b05fc9 100644
--- a/drivers/media/i2c/imx376.c
+++ b/drivers/media/i2c/imx376.c
@@ -1099,13 +1099,19 @@ static int imx376_power_on(struct device *dev)
 
 	ret = clk_prepare_enable(imx376->clk);
 	if (ret) {
-		dev_err(dev, "failed to enable clock\n");
-		regulator_bulk_disable(IMX376_NUM_SUPPLIES, imx376->supplies);
+		dev_err(dev, "failed to enable inclk\n");
+		goto error_reset;
 	}
 
 	usleep_range(1000, 1200);
 
 	return 0;
+
+error_reset:
+	gpiod_set_value_cansleep(imx376->reset_gpio, 1);
+	regulator_bulk_disable(IMX376_NUM_SUPPLIES, imx376->supplies);
+
+	return ret;
 }
 
 static int imx376_power_off(struct device *dev)
-- 
2.49.0

