From 47be2e4e756c6d2fd25a45b0fe3afff5035ff0b6 Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Sun, 9 Feb 2025 19:14:25 +0100
Subject: [PATCH 142/142] fixup media: i2c: Add imx371 image sensor driver

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 drivers/media/i2c/imx371.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/imx371.c b/drivers/media/i2c/imx371.c
index f62fd9d368..f69dd204f8 100644
--- a/drivers/media/i2c/imx371.c
+++ b/drivers/media/i2c/imx371.c
@@ -970,13 +970,18 @@ static int imx371_power_on(struct device *dev)
 
 	ret = clk_prepare_enable(imx371->clk);
 	if (ret) {
-		dev_err(dev, "failed to enable clock\n");
-		regulator_bulk_disable(IMX371_NUM_SUPPLIES, imx371->supplies);
+		dev_err(dev, "failed to enable inclk\n");
+		goto error_reset;
 	}
 
 	usleep_range(1000, 1200);
 
 	return 0;
+error_reset:
+	gpiod_set_value_cansleep(imx371->reset_gpio, 1);
+	regulator_bulk_disable(IMX371_NUM_SUPPLIES, imx371->supplies);
+
+	return ret;
 }
 
 static int imx371_power_off(struct device *dev)
-- 
2.49.0

